Add server_time method to kraken exchange

sdeslauriers opened the pull request:
This commit starts implementing the `Kraken` exchange. For now the only complete method is `server_time`. All other methods are dummies to be able to instantiate and test the new class.

sdeslauriers committed:
Add server_time method to kraken exchange

This commit starts implementing the Kraken exchange. For now
the only complete method is server_time. All other methods are
dummies to be able to instanciate and test the new class.

sdeslauriers committed:
Merge branch 'master' into add-kraken-exchange

gagdiez commented on exchanges/kraken.py:
```diff
@@ -0,0 +1,35 @@
+        # if there is a connection error or if the request times
+        # out. In addition, raise an exception if the status is not
+        # OK.
+        response = requests.get(self.url + 'public/Time', timeout=self.timeout)
```
@sdeslauriers. Maybe we should implement a _public_query method? as in: https://github.com/gagdiez/exchanges/blob/cassandra/exchanges/kraken/api.py. All the public requests to kraken follow the same format.

sdeslauriers replied:
Good idea!

sdeslauriers commented on exchanges/kraken.py:
```diff
@@ -0,0 +1,35 @@
+        # if there is a connection error or if the request times
+        # out. In addition, raise an exception if the status is not
+        # OK.
+        response = requests.get(self.url + 'public/Time', timeout=self.timeout)
```
Good idea.

gagdiez commented:
@sdeslauriers. Maybe we should implement a _public_query method? as in: https://github.com/gagdiez/exchanges/blob/cassandra/exchanges/kraken/api.py. All the public requests to kraken follow the same format.

gagdiez reviewed (commented):
This commit is ok, but maybe we should abstract the public queries to kraken in a method.

sdeslauriers committed:
Add private method for Kraken public data queries

All public data queries of the Kraken API follow the same
pattern. To reduce code duplication, a private method is added to
the Kraken class to get public data.

gagdiez reviewed (approved):
The _public_query method is able to query kraken and handle exceptions. 10/10 would approve.

sdeslauriers commented:
Commit 167ae337fcdc100aee57ef434f9e15277be94987 adds a private method to access public data.
